dashboard/app/mailers/email_delivery_interceptor.rb

```diff
@@ -6,8 +6,8 @@
 #
 class EmailDeliveryInterceptor
   def self.delivering_email(message)
-    action = ActionMailerMetrics.get_message_action(message)
-    mailer_class = ActionMailerMetrics.get_message_class(message)
+    action = ActionMailerMetrics.get_message_action(message) || 'unknown_action'
+    mailer_class = ActionMailerMetrics.get_message_class(message) || 'unknown_class'
```

Comment:
Why not keep this logic inside `ActionMailerMetrics.get_message_action` and `ActionMailerMetrics.get_message_class`?

Reply:
```suggestion
class ActionMailerMetrics
  UNKNOWN_ACTION = 'unknown_action'.freeze
  UNKNOWN_CLASS = 'unknown_class'.freeze
  
  def self.get_message_action(message)
    message_action = message.header[ACTION_HEADER]
    message_action ? message_action.to_s : UNKNOWN_ACTION
  end

  # Returns the ActionMailer class name which created the given message.
  def self.get_message_class(message)
    message_class = message.header[CLASS_HEADER]
    message_class ? message_class.to_s : UNKNOWN_CLASS
  end
end
```

Reply:
> Why not keep this logic inside ActionMailerMetrics.get_message_action and ActionMailerMetrics.get_message_class?

Just a minor reason. I expect a "get*" function to return nil if nothing is defined. This requirement for having a non-empty/nil metric value is related to metrics, so it felt more natural to have `action_name.nil? ? 'unknown' : action_name` closer to where the metrics are logged.

This is a very minor thing though, I will just go with what you have suggested since there are less lines of code.

Reply:
Because of `.to_s` it returns an empty string `nil.to_s == ''`

Reply:
`message.header[CLASS_HEADER].to_s` - `.to_s` converts `nil` to an empty string `''` which is an truthy value `('' || 'unexpected)' == ''`.
If you need to return `message_class` as a string or `nil` then you can do  `message.header[CLASS_HEADER].to_s.presence`

Reply:
```ruby
class ActionMailerMetrics
  UNKNOWN_CLASS = 'unknown_class'.freeze
  
  def self.get_message_class(message)
    message.header[CLASS_HEADER].to_s.presence
  end
end

mailer_class = ActionMailerMetrics.get_message_class(message) || ActionMailerMetrics::UNKNOWN_CLASS
```